src/oc_erchef/apps/chef_license/src/chef_license_worker.erl

```diff
@@ -0,0 +1,128 @@
+-module(chef_license_worker).
+
+-behaviour(gen_server).
+
+-include("../../../include/chef_types.hrl").
+
+-export([
+    start_link/0,
+    get_license/0,
+    test_license/0
+]).
+
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    code_change/2,
+    terminate/2
+]).
+
+-record(state, {
+    scanned_time,
+    grace_period,
+    license_cache,
+    message
+}).
+
+-define(DEFAULT_LICENSE_SCAN_INTERVAL, 3000). %milli seconds
+
+-define(DEFAULT_FILE_PATH, "/tmp/lic").
+
+-define(LICENSE_SCAN_CMD, "chef-automate license status --result-json ").
+
+
+%%% ======================================
+%%% Exported
+%%% ======================================
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+get_license()->
+    gen_server:handle_call(?MODULE, get_license).
+
+%%% ======================================
+%%% Gen Server callbacks
+%%% ======================================
+init(_Config) ->
+    State = check_license(#state{}),
+    gen_server:send_after(?DEFAULT_LICENSE_SCAN_INTERVAL, check_license),
```

Comment:
This is better as a direct function call to `check_license/1`.  Otherwise, there's a gap where the gen_server is not in the right state (between init, and when that first check_license runs.  By moving the check_license function call here, you can guarantee either a restart on failure;  or a valid state after `init` completion.